swarm/pss/pss.go

```diff
@@ -294,12 +300,14 @@ func (self *Pss) handlePssMsg(msg interface{}) error {
 				return errors.New("Invalid TTL")
 			}
 			log.Trace("pss was for someone else :'( ... forwarding", "pss", common.ToHex(self.BaseAddr()))
-			return self.forward(pssmsg)
+			self.checkFwdCache([]byte{}, digest, true)
+			return self.forward(pssmsg, digest)
 		}
 		log.Trace("pss for us, yay! ... let's process!", "pss", common.ToHex(self.BaseAddr()))
 
-		if !self.process(pssmsg) {
-			err = self.forward(pssmsg)
+		if !self.process(pssmsg, digest) {
+			self.checkFwdCache([]byte{}, digest, true)
```

Comment:
The same as above https://github.com/ethersphere/go-ethereum/pull/314/files#diff-2d1fd9a8750299d8ae63031e6b30be89R303.